wallet/src/main/scala/org/bitcoins/wallet/api/WalletApi.scala

```diff
@@ -330,30 +329,17 @@ trait LockedWalletApi extends WalletApi {
         }
       }
 
-      /** Iterates over all filters in the range to find matches */
-      @tailrec
-      def loop(
-          start: Int,
-          end: Int,
-          acc: Future[Vector[BlockMatchingResponse]]): Future[
-        Vector[BlockMatchingResponse]] = {
-        if (end <= start) {
-          acc
-        } else {
-          val startHeight = end - (batchSize - 1)
-          val endHeight = end
-          val newAcc = for {
-            compactFilterDbs <- chainQueryApi.getFiltersBetweenHeights(
-              startHeight,
-              endHeight)
-            filtered <- findMatches(compactFilterDbs)
-            res <- acc
-          } yield {
-            res ++ filtered
-          }
-          val newEnd = Math.max(start, endHeight - batchSize)
-          loop(start, newEnd, newAcc)
-        }
+
+
+      def fetchFiltersInRange(heightRange: Vector[Int]): Future[Vector[BlockMatchingResponse]] = {
```

Comment:
I think this would be better as `def fetchFiltersInRange(startHeight: Int, stopHeight: Int)`

Reply:
if not then you should do
```
val startHeight = heightRange.min
val endHeight = heightRange.max
```